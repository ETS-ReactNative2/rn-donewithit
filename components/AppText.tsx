import React, { useMemo } from 'react';
import { Text, StyleSheet, Platform } from 'react-native';

const AppText = (props: any) => {
  const styles = useMemo(()=>(StyleSheet.create({
    text: {
      // ...Platform.select({
      //   android: {
      //     fontSize: 20,
      //     fontFamily: 'Roboto',
      //     fontWeight: 'bold',
      //   },
      //   ios: {
      //     fontSize: 18,
      //     fontFamily: 'Courier',
      //     fontWeight: '100',
      //   }
      // }),
      // ==============================
      fontStyle: 'italic',
      color: 'tomato',
      textTransform: 'capitalize',
      textDecorationLine: 'underline',
      textAlign: 'center',
      lineHeight: 36,
    }
  })), [])

  return (
    <Text {...props} style={styles.text}>{props.children}</Text>
  );
}

export default AppText
