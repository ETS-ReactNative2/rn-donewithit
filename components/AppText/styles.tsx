import { StyleSheet, Platform } from 'react-native'
import COLORS from '../../config/colors'
const styles = StyleSheet.create({
  text: {
    ...Platform.select({
      android: {
        fontFamily: 'Roboto'
      },
      ios: {
        fontFamily: 'Courier'
      }
    }),
    fontSize: 13,
    color: COLORS.BLACK
  }
})

export default styles
