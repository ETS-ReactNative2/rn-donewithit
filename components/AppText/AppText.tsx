import React from 'react'
import { Text } from 'react-native'
import styles from './styles'

function AppText({ style, children, ...otherProps }: any) {
  return (
    <Text style={[styles.text, style]} {...otherProps}>
      {children}
    </Text>
  )
}

export default AppText
