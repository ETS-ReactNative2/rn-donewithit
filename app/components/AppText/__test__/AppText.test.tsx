import React from 'react'
import { StyleSheet } from 'react-native'
import renderer from 'react-test-renderer'
import AppText from '../index'

const customStyles = StyleSheet.create({
  text: {
    fontSize: 16,
    color: 'red'
  }
})

const props = {
  children: 'Hello world',
  style: customStyles.text
}

describe('AppText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppText {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
