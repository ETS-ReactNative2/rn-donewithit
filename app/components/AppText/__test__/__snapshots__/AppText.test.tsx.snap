// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText renders correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
        "fontFamily": "Courier",
        "fontSize": 13,
      },
      Object {
        "color": "red",
        "fontSize": 16,
      },
    ]
  }
>
  Hello world
</Text>
`;
